docs/application/dotnet/guides/machine-learning/machine-learning-train.md

```diff
@@ -0,0 +1,778 @@
+```
+[[3072 pixels][20 floats for coarse label][100 floats for fine label]],
+...
+...
+[[3072 pixels][20 floats for coarse label][100 floats for fine label]]
+```
+
+After preparing the file, create dataset as follows:
+
+```csharp
+
+string datasetFilePath = Tizen.Applications.Application.Current.DirectoryInfo.Data + "trainingSet.dat";
+
+try
+{
+  var model = new Model();
+  var dataset = new Dataset();
+  // a file data provider to be used for training
+  dataset.AddFile(NNTrainerDatasetMode.Train, datasetFilePath);
+  dataset.SetProperty(NNTrainerDatasetMode.Train, "buffer_size=9");
+
+  model.SetDataset(dataset);
+}
+catch (Exception e)
+{
+  // handle error
+}
+```
+The property `buffer_size` defines the maximum amount of batches to be queued while training. When it is not given, it is set to 1.
+
+## Compile the model
+
+Compiling a model finalizes the model with loss.
+Once compiled, any modification to the properties of the model is restricted.
+Adding layers or changing the optimizer or dataset of the model is not permitted either:
+
+```csharp
+try
+{
+  model.Compile("loss=cross");
+}
+catch (Exception e)
+{
+  // handle error
+}
+```
+
+## Train the model
+
+Now, the model is ready to train. Train model as follows:
+
+```csharp
+try
+{
+  model.Run("batch_size=9", "epochs=20");
+}
+catch (Exception e)
+{
+  // handle error
+}
+```
+## Destroy the model
+
+Model will be disposed by GC but you can use Dispose().
+`Tizen.MachineLearning.Train.Model.AddLayer`, `Tizen.MachineLearning.Train.Model.SetOptimizer`, and `Tizen.MachineLearning.Train.Model.SetDataset` transfers ownership to the model.
+`Tizen.MachineLearning.Train.Layer`, `Tizen.MachineLearning.Train.Optimizer` and `Tizen.MachineLearning.Train.Dataset` that belongs to the `model` are also deleted.
+
+## Use the trained model for inference
+
+The trained model can be used for inference with [Machine Learning Inference API](machine-learning-inference.md).
+Ensure that the INI file contains the correct weight file in `save_path` in `[Model]` section.
+The valid INI file can be made with`Tizen.MachineLearning.Train.Model.Save` if you have constructed the model with the provided method.
+For example, you can use the trained model with a single API as follows:
+
+```csharp
+using Tizen.MachineLearning.Inference;
+
+TensorInfo inInfo;
+TensorInfo outInfo;
+string modelFilePath = Tizen.Applications.Application.Current.DirectoryInfo.Data + "modelFile.ini";
+...
+
+var single = new SingleShot(modelFilePath, inInfo, outInfo);
+```
+
+You can use the trained model with pipeline API as follows:
+
+```csharp
+using Tizen.MachineLearning.Inference;
+
+/* framework is 'nntrainer'*/
+const string description = "appsrc ! other/tensor,dimension=(string)2:1:1:1,type=(string)  int8,framerate=(fraction)0/1 ! tensor_filter framework=nntrainer model=model.ini ! tensor_sink";
+
+var pipeline = new Pipeline(description);
+```
```

Comment:
This md file (`machine-learning-train.md`) is missing a Related Information section mentioning dependencies.